compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3923,12 +3924,27 @@ abstract class GenJSCode extends plugins.PluginComponent
           }
         }
 
+        def keyToPropName(key: js.Tree): js.PropertyName = key match {
+          case key: js.StringLiteral =>
+            key
+          case _ =>
+            /* ComputedName wants a logicalName, but it is not used for
+             * JSObjectConstr.
+             * TODO We should improve this.
+             */
+            js.ComputedName(key, "dummy")
```

Comment:
Calling the logical name "dummy" is OK-ish (I would suggest "local"). However, the `ComputeName` names should be unique, so we should probably add a counter or something.